group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -2437,6 +2438,16 @@ private CoordinatorResult<Void, CoordinatorRecord> classicGroupJoinToConsumerGro
             records
         );
 
+        bumpGroupEpoch |= maybeUpdateRegularExpressions(
+            context,
+            group,
+            member,
+            updatedMember,
+            records
+        );
+
+        bumpGroupEpoch |= !member.subscribedTopicRegex().equals(updatedMember.subscribedTopicRegex());
```

Comment:
Could we put a comment to explain the rational of this one? I also wonder whether we could just check that the previous one was non-empty as it is a simple check. What do you think?